test/cli/credentials/list.test.js

```diff
@@ -1,58 +1,187 @@
+    const types = {};
+    const keyAuthKeys = [];
+    env.hijack(namespace, generator => {
+      generator.once('run', () => {
+        generator.log.error = message => {
+          done(new Error(message));
+        };
+        generator.stdout = msg => {
+          const crd = JSON.parse(msg);
+          types[crd.type] = (types[crd.type] || 0) + 1;
+          if (crd.type === 'key-auth') {
+            keyAuthKeys.push(crd.keyId);
+          }
+        };
+      });
+
+      generator.once('end', () => {
+        keyAuthKeys.sort();
+        createdKeyAuthKeys.all.sort();
+
+        assert.deepEqual(types, createdTypes.active);
+        assert.deepEqual(keyAuthKeys, createdKeyAuthKeys.active);
+        done();
+      });
     });
+
+    env.argv = program.parse('credentials list -c ' + username);
   });
 
-  it('should show credentials', done => {
-    let output = {};
+  it('should show all archive credentials', done => {
+    const types = {};
+    const keyAuthKeys = [];
     env.hijack(namespace, generator => {
       generator.once('run', () => {
         generator.log.error = message => {
           done(new Error(message));
         };
         generator.stdout = msg => {
-          let crd = JSON.parse(msg);
-          output[crd.type] = (crd.keyId || crd.secret || crd.password);
+          const crd = JSON.parse(msg);
+          types[crd.type] = (types[crd.type] || 0) + 1;
+          if (crd.type === 'key-auth') {
+            keyAuthKeys.push(crd.keyId);
+          }
         };
       });
 
       generator.once('end', () => {
-        assert.ok(output['oauth2']);
-        assert.ok(output['basic-auth']);
-        assert.equal(output['key-auth'], keyCred1.keyId);
+        keyAuthKeys.sort();
+        createdKeyAuthKeys.archive.sort();
+
+        assert.deepEqual(types, createdTypes.archive);
+        assert.deepEqual(keyAuthKeys, createdKeyAuthKeys.archive);
+        done();
+      });
+    });
+
+    env.argv = program.parse('credentials list -i archive -c ' + username);
+  });
+
+  it('should show all active and archive credentials', done => {
+    const types = {};
+    const keyAuthKeys = [];
+    env.hijack(namespace, generator => {
+      generator.once('run', () => {
+        generator.log.error = message => {
+          done(new Error(message));
+        };
+        generator.stdout = msg => {
+          const crd = JSON.parse(msg);
+          types[crd.type] = (types[crd.type] || 0) + 1;
+          if (crd.type === 'key-auth') {
+            keyAuthKeys.push(crd.keyId);
+          }
+        };
+      });
+
+      generator.once('end', () => {
+        keyAuthKeys.sort();
+        createdKeyAuthKeys.all.sort();
+
+        assert.deepEqual(types, createdTypes.all);
+        assert.deepEqual(keyAuthKeys, createdKeyAuthKeys.all);
         done();
       });
     });
 
-    env.argv = program.parse('credentials list -c ' + user.username);
+    env.argv = program.parse('credentials list -i archive,active -c ' + username);
```

Comment:
instead of credentials list -i archive,active -c  username
it should be 
credentials list -i archive -i active -c  username